coalib/bearlib/abstractions/Linter.py

```diff
@@ -0,0 +1,680 @@
+                    properties.
+                :param severity_map:
+                    A dict used to map a severity string (captured from the
+                    ``output_regex`` with the named group ``severity``) to an
+                    actual ``coalib.results.RESULT_SEVERITY`` for a result.
+                :return:
+                    An iterator returning results.
+                """
+                if isinstance(output, str):
+                    output = (output,)
+
+                for string in output:
+                    for match in re.finditer(output_regex, string):
+                        yield self._convert_output_regex_match_to_result(
+                            match, filename, severity_map=severity_map)
+
+            if options["output_format"] is None:
+                # Check if user supplied a `process_output` override.
+                if not (hasattr(klass, "process_output") and
+                        callable(klass.process_output)):
+                    raise ValueError("`process_output` not provided by given "
+                                     "class.")
+                # No need to assign to `process_output` here, the class mixing
+                # below automatically does that.
+            else:
+                # Prevent people from accidentally defining `process_output`
+                # manually, as this would implicitly override the internally
+                # set-up `process_output`.
+                if hasattr(klass, "process_output"):
+                    raise ValueError("`process_output` is used by given class,"
+                                     " but " + repr(options["output_format"]) +
+                                     " output format was specified.")
+
+                if options["output_format"] == "corrected":
+                    process_output_args = {}
+                    if "diff_severity" in options:
+                        process_output_args["diff_severity"] = (
+                            options["diff_severity"])
+                    if "diff_message" in options:
+                        process_output_args["diff_message"] = (
+                            options["diff_message"])
+
+                    process_output = partialmethod(
+                        process_output_corrected, **process_output_args)
+
+                elif options["output_format"] == "regex":
+                    process_output_args = {
+                        "output_regex": options["output_regex"]}
+                    if "severity_map" in options:
+                        process_output_args["severity_map"] = (
+                            options["severity_map"])
+
+                    process_output = partialmethod(
+                        process_output_regex, **process_output_args)
+
+                else:  # pragma: no cover
+                    # This statement is never reached.
+                    assert False
```

Comment:
Actually, it can be reached if a invalid output_format is given ...
So: `raise ValueError("Invalid value for output_format was found")` makes more sense ?